Helper/ClientManager.php

```diff
@@ -54,26 +50,38 @@ public function __construct(
         $this->moduleList = $moduleList;
     }
 
-    public function initClient($storeId = null)
+    public function getClient($storeId = null)
     {
-        $apiKey = $this->getSettingsVal(self::XML_API_KEY, $storeId);
-        $apiSecret = $this->getSettingsVal(self::XML_API_SECRET, $storeId);
-        $this->client = new MageClient($apiKey, $apiSecret, $this->getSetupVersion(), $this->logger, $this->scopeResolver);
+        $cacheKey = (int)$storeId;
+        if (!isset($this->clientsRegistry[$cacheKey])) {
+            $this->clientsRegistry[$cacheKey] = $this->initClient($storeId);
+        }
+
+        return $this->clientsRegistry[$cacheKey];
     }
 
-    public function getClient($update=false, $storeId = null)
+    /**
+     * @param null|int|string $storeId
+     *
+     * @return MageClient
+     */
+    public function initClient($storeId = null)
     {
-        if ($update or !$this->client) {
-            $this->initClient($storeId);
-        }
-        return $this->client;
+        return new MageClient(
+            $this->getSettingsVal(self::XML_API_KEY, $storeId),
```

Comment:
Reviewing the code for `getSettingsVal`, I'm not convinced it works for a null $storeId. Do you have any concerns?

Reply:
Here I just refactored code style and overall method logic. Store ID in entire Sailthru integration logic shouldn't be null